test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua

```diff
@@ -0,0 +1,70 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0041-appicon-resumption.md
+-- User story:TBD
+-- Use case:TBD
+--
+-- Requirement summary:
+-- TBD
+--
+-- Description:
+-- In case:
+-- 1) SDL, HMI are started.
+-- 2) Mobile application is registered and sets custom icon via sending PutFile and valid SetAppIcon request.
+-- 3) Mobile application is unregistered.
+-- 4) Mobile app is re-registered.
+-- SDL does:
+-- 1) Respond with result code "SUCCESS" and "iconResumed" = true for RAI request.
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/SetAppIcon/commonIconResumed')
+local test = require("user_modules/dummy_connecttest")
+local mobile_session = require('mobile_session')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local requestParams = {
+  syncFileName = "icon.png"
+}
+local requestUiParams = {
+  syncFileName = {
+    imageType = "DYNAMIC",
+    value = common.getPathToFileInStorage(requestParams.syncFileName)
+  }
+}
+local allParams = {
+  requestParams = requestParams,
+  requestUiParams = requestUiParams
+}
+
+local function CloseConnection()
+	test.mobileConnection:Close()
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
```

Comment:
@HSavynetska For better stability of the test expectation should precede the action